pkg/image/validation.go

```diff
@@ -0,0 +1,153 @@
+package image
+
+import (
+	"fmt"
+	"strings"
+)
+
+func validate(definition *Definition) error {
+	err := validateAPIVersion(definition)
+	if err != nil {
+		return fmt.Errorf("error validating apiVersion: %w", err)
+	}
+	err = validateImage(definition)
+	if err != nil {
+		return fmt.Errorf("error validating image: %w", err)
+	}
+	err = validateOperatingSystem(definition)
+	if err != nil {
+		return fmt.Errorf("error validating operating system: %w", err)
+	}
+
+	return nil
+}
+
+func validateAPIVersion(definition *Definition) error {
+	// Open to discussion regarding how to validate the else if
+	if definition.APIVersion == (Definition{}.APIVersion) {
+		return fmt.Errorf("apiVersion not defined")
+	} else if len(definition.APIVersion) < 2 {
+		return fmt.Errorf("invalid apiVersion")
+	}
+	return nil
+}
+
+func validateImage(definition *Definition) error {
+	if definition.Image == (Image{}) {
```

Comment:
This is fine and will work for the time being. There's always the opportunity to introduce better checks by using pointers to structs instead of structs but I wouldn't spend the time on such enhancements right now.